mlflow/spark.py

```diff
@@ -385,6 +407,30 @@ def maybe_copy_from_uri(cls, src_uri, dst_path):
     def delete(cls, path):
         cls._fs().delete(cls._remote_path(path), True)
 
+    @classmethod
+    def is_filesystem_available(cls, scheme):
+        return scheme in [stats.getScheme() for stats in cls._stats().iterator()]
+
+
+def _should_use_mlflowdbfs(root_uri):
+    # The `mlflowdbfs` scheme does not appear in the available schemes returned from
+    # the Hadoop FileSystem API until a read call has been issued.
+    from mlflow.utils._spark_utils import _get_active_spark_session
+
+    try:
+        _get_active_spark_session().read.load("mlflowdbfs:///artifact?run_id=foo&path=/bar")
+    except Exception:
+        # The load invocation is expected to throw an exception.
+        pass
+
+    return (
+        is_valid_dbfs_uri(root_uri)
+        and is_databricks_acled_artifacts_uri(root_uri)
+        and databricks_utils.is_in_databricks_runtime()
+        and not environment_variables.DISABLE_MLFLOWDBFS.get()
```

Comment:
If someone sets the value `false` or `False` for `DISABLE_MLFLOWDBFS` env var, looks like we enable mlflowdbfs anyway because `false` / `False` are truthy strings. Can we do some string matching to fix this and add a test?